src/main/java/ladder/domain/Record.java

```diff
@@ -0,0 +1,48 @@
+package ladder.domain;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class Record {
+  private final Map<Player, Integer> values;
+
+  private Record(Map<Player, Integer> values) {
+    this.values = Collections.unmodifiableMap(values);
+  }
+
+  public static Record generate(final Players players, final Ladder ladder) {
+    final List<Player> playersValues = players.getValues();
+
+    Map<Player, Integer> result = playersValues.stream()
+            .collect(Collectors.toMap(Function.identity(), player -> ladder.getLastDepthStartLineIndex(playersValues.indexOf(player))));
+    return new Record(result);
+  }
```

Comment:
```suggestion
  private Record(Map<Player, Integer> values) {
    this.values = values;
  }

  public static Record generate(final Players players, final Ladder ladder) {
    final List<Player> playersValues = players.getValues();

    Map<Player, Integer> result = playersValues.stream()
            .collect(Collectors.toMap(Function.identity(), player -> ladder.getLastDepthStartLineIndex(playersValues.indexOf(player))));
    return new Record(Collections.unmodifiableMap(result));
  }
```

values 를 불변 컬렉션으로 만드는 책임은 생성자가 아니라, generate 에 있다고 생각됩니다.
생성자의 역할은 받아온 values 가 불변이던, 가변이던 상관없이 그대로 values 에 할당하는 역할이라고 생각이 들어서요.
생성자에게는 Exception 을 throw 하는 것 외에는 로직처리를 최소화 하면 좋을 것 같습니다.